### PR TITLE
OCP-24818 Checking OLM descriptors

### PR DESCRIPTION
**Pull request by tbuskey**
oc explain for InstallPlan, ClusterServiceVersion, Subscription, CatalogSource, OperatorSource, OperatorGroup, PackageManifest, CatalogSourceConfig should not be empty

Test case: [https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-24818](url)

Succeeds on OCP 4.4, fails 5(?) on OCP 4.3

If any one of the descriptors is empty, the tests fails and logs the ones that failed and a total count
`./extended-platform-tests run all --dry-run | grep 'Checking OLM descriptors' | ./extended-platform-tests run -f - 2>&1 > tee testrun`
[testrun-24818.log](https://github.com/openshift/openshift-tests/files/4221051/testrun-24818.log)

### CODE REVIEW

File: test/extended/operators/olm.go [hunk 1]
```diff
@@ -226,4 +226,28 @@ var _ = g.Describe("[Feature:Platform] an end user use OLM", func() {
 		}
 
 	})
+
+	// OCP-24818 - Checking OLM descriptors
+	// author: tbuskey@redhat.com
+	g.It("Checking OLM descriptors", func() {
+		olmErr := 0
+		olmErrDescriptor := []string{""}
+		olmExplains := []string{"InstallPlan", "ClusterServiceVersion", "Subscription", "CatalogSource", "OperatorSource", "OperatorGroup", "PackageManifest", "CatalogSourceConfig"}
```
Comment (awgreene) [inline test/extended/operators/olm.go]: CatalogSourceConfig is not an OLM resource, it belongs to the Marketplace Operator.

File: test/extended/operators/olm.go [hunk 2]
```diff
@@ -226,4 +226,28 @@ var _ = g.Describe("[Feature:Platform] an end user use OLM", func() {
 		}
 
 	})
+
+	// OCP-24818 - Checking OLM descriptors
+	// author: tbuskey@redhat.com
+	g.It("Checking OLM descriptors", func() {
```
Comment (awgreene) [inline test/extended/operators/olm.go:232]: Is this being done for all CRDs shipped with OpenShift?

**Review by awgreene — COMMENTED** [review]
2 small comments - thanks!

**Comment by awgreene** [comment]
>Succeeds on OCP 4.4, fails 5(?) on OCP 4.3

@tbuskey  this will continue to fail on 4.3 until https://github.com/operator-framework/operator-lifecycle-manager/pull/1197 is merged

**Comment by tbuskey** [comment]
@awgreene Yes, this fails on 4.3, but it demonstrates the log output.

**Comment by tbuskey** [comment]
Removed CatalogSourceConfig check

File: test/extended/operators/olm.go (hunk 2 above)
Comment (tbuskey) [reply test/extended/operators/olm.go:232]: @jianzhangbjz Do you have direction here?

Comment (awgreene) [reply test/extended/operators/olm.go:232]: This test seems extremely similar to those featured in https://github.com/openshift/origin/pull/22696 as well. Could you explain how the tests featured in openshift/origin are different from those featured in the openshift/openshift-test repository?

Comment (tbuskey) [reply test/extended/operators/olm.go:232]: We were directed to move QE OLM tests out of the origin repo to this one.

Comment (awgreene) [reply test/extended/operators/olm.go:232]: Okay - that's good to know - thanks!

Comment (jianzhangbjz) [reply test/extended/operators/olm.go:232]: Yes, we're migrating the extended tests from origin to this repo. Details in: https://github.com/openshift/enhancements/pull/183

**Pull request merged**